src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -153,18 +161,51 @@ const peerConnectRequestSignChangeHandler = async (
   );
 };
 
+const peerConnectedChangeHandler = async (
+  event: PeerConnectedEvent,
+  dispatch: ReturnType<typeof useAppDispatch>
+) => {
+  const peerConnection =
+    await Agent.agent.peerConnectionMetadataStorage.getPeerConnectionMetadata(
+      event.payload.dAppAddress
+    );
+  const existingConnections =
+    await Agent.agent.peerConnectionMetadataStorage.getAllPeerConnectionMetadata();
```

Comment:
I think we can update redux with the peer connection as soon as we click "Connect" when selecting the identifier, and not wait until now. Can we do that there, and without this extra call to the DB to get all connects? Would be nicer to just append.

So then the only thing we update here is which one is connected fully (`setConnectedWallet`).

Reply:
Yes but I'm only talking about fetching from the DB, not writing to it. We already have Redux updates to update with the correct information.

We should be able to do a Redux update to add it initially too when we click connect (even if it doesn't have much information), and then once the information comes back, update Redux again for that connection.

This part in AppWrapper is about the connection becoming connected - which should just update the Redux for the connected item.

Reply:
Don't we still need to fetch the DB to get the latest version of the newly connected record?

Reply:
But can we not update Redux at the same time we are writing that information to the DB?

I am suggesting we update Redux whenever we are writing to the DB, instead of fetching from the DB here.

Reply:
(I don't mean write to redux from the core either, I just mean whenever we call that function from the UI)

Reply:
Hmmm. Actually, the writing happens before the update, I am using `await` so it is not at the same time. Is this what you are concerned about?

Reply:
agreed on slack to keep the flow as is for simplicity